File: samples/RazorLibrary/ExampleRazorLibraryVB/App_Start/RazorGeneratorMvcStart.vb
Imports System.Web.WebPages
Imports RazorGenerator.Mvc

<Assembly: WebActivatorEx.PostApplicationStartMethod(
    GetType(RazorSystemWeb.RazorGeneratorMvcStart),
    NameOf(RazorSystemWeb.RazorGeneratorMvcStart.Start)
    )>

Namespace RazorSystemWeb
    Public NotInheritable Class RazorGeneratorMvcStart
        Private Sub New()

        End Sub

        Public Shared Sub Start()
            Dim engine = New PrecompiledMvcEngine(GetType(RazorGeneratorMvcStart).Assembly) With {
                .UsePhysicalViewsIfNewer = HttpContext.Current.Request.IsLocal
            }

            ViewEngines.Engines.Insert(0, engine)

            '' StartPage lookups are done by WebPages. 
            VirtualPathFactoryManager.RegisterVirtualPathFactory(engine)
        End Sub
    End Class
End Namespace

File: src/MSBuild.SDK.SystemWeb.Templates/templates/RazorLibrary.SystemWeb.CS/App_Start/RazorGeneratorMvcStart.cs
﻿using System.Web;
using System.Web.Mvc;
using System.Web.WebPages;
using RazorGenerator.Mvc;

[assembly: WebActivatorEx.PostApplicationStartMethod(
    typeof(ExampleRazorLibrary.RazorGeneratorMvcStart), 
    nameof(ExampleRazorLibrary.RazorGeneratorMvcStart.Start)
    )]

namespace ExampleRazorLibrary
{
    public static class RazorGeneratorMvcStart
    {
        public static void Start()
        {
            var engine = new PrecompiledMvcEngine(typeof(RazorGeneratorMvcStart).Assembly)
            {
                UsePhysicalViewsIfNewer = HttpContext.Current.Request.IsLocal
            };

            ViewEngines.Engines.Insert(0, engine);

            // StartPage lookups are done by WebPages. 
            VirtualPathFactoryManager.RegisterVirtualPathFactory(engine);
        }
    }
}
